### PR TITLE
Disable generation of default menu item name if name already exists.

### PR DESCRIPTION
**Pull request by bereznev**
…xists.

### CODE REVIEW

**Comment by LukeTowers** [comment]
Please provide replication instructions for the issue that this solves.

**Comment by bereznev** [comment]
When I add a new item menu of static page, it name is the default name "Nová položka menu" instead of page name.

**Comment by LukeTowers** [comment]
@bereznev are you able to provide a GIF or video screen recording of the issue? That would help a lot in replicating the issue.

**Comment by mjauvin** [comment]
I can't reproduce this. what version of Pages plugin and Translate plugin do you use @bereznev?

**Comment by bereznev** [comment]
ezgif-6-aba7f4e27cb8

**Comment by bereznev** [comment]
Expected:
ezgif-7-6bec3f414c64

**Comment by mjauvin** [comment]
Ok, yes, I was able to reproduce. This is definitely a bug.

Your fix does resolve the issue.

File: formwidgets/menuitems/assets/js/menu-items-editor.js [hunk 1]
```diff
@@ -418,7 +418,7 @@
             // If the RainLab.Translate default locale data locker fields are available make sure the regular inputs are properly populated
             if (defaultLocale) {
                 var $defaultLocaleField = $('[name="RLTranslate['+defaultLocale+']['+propertyName+']"]', self.$popupContainer)
```
Comment (LukeTowers) [inline formwidgets/menuitems/assets/js/menu-items-editor.js:420]:
```suggestion
            if (defaultLocale && !$input.val()) {
                var $defaultLocaleField = $('[name="RLTranslate['+defaultLocale+']['+propertyName+']"]', self.$popupContainer)
```

**Comment by LukeTowers** [comment]
Arg, github screwed up. Can you make my recommended change in your branch?

**Comment by bennothommo** [comment]
While this fix does somewhat fix the issue, I don't think it's quite the right fix.

The issue is that the default language's data locker value is not being set with the selected reference's title - see image below:

Screenshot (6)

The hidden field `RLTranslate[en][title]` shows "New menu item", regardless of the correct title showing in the main input. You'll notice that if you switch the language to another language, then switch it back, "New menu item" will re-appear in the main input.

If you can get that data locker value using the correct title, I think that'll properly fix the issue.

**Comment by mjauvin** [comment]
This is where it needs to be added:

https://github.com/rainlab/pages-plugin/blob/1929a484a5f299c600e9137d4c6bbfb5fda43755/formwidgets/menuitems/assets/js/menu-items-editor.js#L89,L96

**Comment by mjauvin** [comment]
Ok, I've got the proper fix for this, I'll commit to your repo.

**Comment by mjauvin** [comment]
@bereznev I created a PR that properly fixes this in your repo.

Thanks.

File: formwidgets/menuitems/assets/js/menu-items-editor.js [hunk 2]
```diff
@@ -91,7 +91,13 @@
...
                 // removing CMS page [base file name] suffix
                 if (selectedTitle && self.properties.title === self.$popupForm.attr('data-new-item-title')) {
-                    $titleField.val($.trim(selectedTitle.replace(/\s*\[.*\]$/, '')))
+                    var title = $.trim(selectedTitle.replace(/\s*\[.*\]$/, ''))
+                    var defaultLocale = $('[data-control="multilingual"]').data('default-locale')
+                    $titleField.val(title)
+                    if (defaultLocale) {
+                        var $localeField = $('[name="RLTranslate['+defaultLocale+'][title]"]', self.$popupContainer)
+                        $localeField.val(title)
+                    }
```
Comment (LukeTowers) [inline formwidgets/menuitems/assets/js/menu-items-editor.js:100]: @mjauvin can you explain what's going on here? I don't quite follow

Comment (mjauvin) [reply formwidgets/menuitems/assets/js/menu-items-editor.js:100]: @LukeTowers when a reference page is selected, it sets the menu item title from the referenced page title.

this extra code also assigns the title to the RainLab.Translate default locale data locker

**Comment by mjauvin** [comment]
@bereznev can you confirm this is completely resolving your issues?

**Comment by bereznev** [comment]
@mjauvin yes thanks, this completely resolved issue

**Comment by mjauvin** [comment]
@bennothommo this is completed and has been tested working by @bereznev and myself.

**Comment by mjauvin** [comment]
@bennothommo can this be added to the next release, Milestone v1.3.7 I believe?

Thanks.

File: formwidgets/menuitems/assets/js/menu-items-editor.js (hunk 2 above)
Comment (LukeTowers) [inline formwidgets/menuitems/assets/js/menu-items-editor.js:100]:
```suggestion
                    var title = $.trim(selectedTitle.replace(/\s*\[.*\]$/, ''))
                    $titleField.val(title)
                    
                    // Support for RainLab.Translate
                    var defaultLocale = $('[data-control="multilingual"]').data('default-locale')
                    if (defaultLocale) {
                        $('[name="RLTranslate['+defaultLocale+'][title]"]', self.$popupContainer).val(title)
                    }
```

**Pull request merged**